hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieLogFormatReader.java

```diff
@@ -46,14 +49,26 @@
     this.readBlocksLazily = readBlocksLazily;
     this.reverseLogReader = reverseLogReader;
     this.bufferSize = bufferSize;
+    this.prevReadersInOpenState = new ArrayList<>();
```

Comment:
@bvaradar We cannot have a list of these readers here. Each reader opens a BufferedInputStream, in the event of a large number of readers (for many log files), this results in OOM.

Reply:
Let me take a look at the code again to see what's the best solution here.

Reply:
I don't think we throw away the readers during lazy read, no?

Reply:
@n3nash : 
W.r.t many log files, I dont think this would make things worse. The reason is currently we anyway queue HoodieLogBlocks (AbstractHoodieLogRecordScanner.scan) which also has an handle to the same buffered-inputstream. 

There is no clear ownership of the input-stream shared by HoodieLogFileReader and all instances of HoodieLogBlocks constructed by it.   Without clear ownership (before this change), the HoodieLogFormatReader only explicitly closes one HoodieLogFileReader instance corresponding to last files. HoodieLogFileReader instances corresponding to other log-files are at the mercy of garbage-collector (and also assuming the underlying input-stream implements finalize correctly).  HoodieLogFormatReader just like java.io.Reader needs a clean semantics around close() which ensures all resources acquired by it are released and not rely on garbage-collection.

Reply:
Actually, I realized why this had resulted in OOM's earlier, it was because we used to have a lastSeenBlock as part of the HoodieLogReader and accumulating Readers was causing OOM's there. I checked that HoodieLogReader does not have the lastSeenBlock any more so it is safe to do this change. 
The ownership of input-stream is purposefully left that way, I believe there is a document around this as to why this choice was taken, I can share with you offline if you're interested. 
Thanks for implementing the close() in the WrapperReader, change looks fine to me.